lib/tests/streamlit/scriptrunner/script_runner_test.py

```diff
@@ -100,11 +101,22 @@ def test_run_script(self, filename, text):
         # work correctly. The CodeHasher is scoped to
         # files contained in the directory of __main__.__file__, which we
         # assume is the main script directory.
-        self.assertEqual(
-            scriptrunner._report.script_path,
-            sys.modules["__main__"].__file__,
-            (" ScriptRunner should set the __main__.__file__" "attribute correctly"),
-        )
+        def __on_event(event):
+            if event == ScriptRunnerEvent.SCRIPT_STOPPED_WITH_SUCCESS:
+                self.assertEqual(
+                    scriptrunner._report.script_path,
+                    sys.modules["__main__"].__file__,
+                    (
+                        " ScriptRunner should set the __main__.__file__"
+                        "attribute correctly"
+                    ),
+                )
+
+        scriptrunner.on_event.connect(__on_event)
+
+        # `sys` should not contain reference to script variables
+        # otherwise GC can not clean up the memory.
+        self.assertNotIn("__main__", sys.modules)
```

Comment:
Does this feel like it should be in a separate test? Seems like a nice simple chunk to test.

Reply:
I had the same feeling :) No problem. Let's move to another test case 👍